main/src/library/Iterator.flix

```diff
@@ -195,108 +174,84 @@ namespace Iterator {
     ///
     /// Consumes the entire iterator.
     ///
-    pub def toChain(iter: Iterator[a, r]): Chain[a] \ Read(r) =
+    pub def toChain(iter: Iterator[a, ef, r]): Chain[a] \ { ef, Read(r) } =
         foldLeft((acc, a) -> Chain.snoc(acc, a), Chain.empty(), iter)
 
     ///
     /// Returns the contents of `iter` as a map.
     ///
     /// Consumes the entire iterator.
     ///
-    pub def toMap(iter: Iterator[(a, b), r]): Map[a, b] \ Read(r) with Order[a] =
-        foldLeft((acc, ab) -> Map.insert(fst(ab), snd(ab), acc), Map.empty(), iter)
+    pub def toMap(iter: Iterator[(k, v), ef, r]): Map[k, v] \ { ef, Read(r) } with Order[k] =
+        foldLeft((acc, a) -> {let (k, v) = a; Map.insert(k, v, acc)}, Map.empty(), iter)
```

Comment:
I believe a match lambda would work here.

Reply:
I missed this - I'll take another look at it.

Reply:
Changed in next commit.

Reply:
Now use match lambda.